File: shortest_unsorted_continuous_subarray/soyeon/ShortestUnsortedContinuousSubarray.cpp
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        vector<int> rightOrder(nums);
        sort(rightOrder.begin(), rightOrder.end());
        
        if(rightOrder == nums) {
            return 0;
        }
        
        int first, last;
        
        for(int i=0; i<rightOrder.size(); i++) {
            cout << rightOrder[i] << ' ';
        }
        
        for(int i=0; i<nums.size(); i++) {
            if(rightOrder[i] != nums[i]) {
                first = i;
                break;
            }
        }
        
        for(int i=nums.size()-1; i>=0; i--) {
            if(rightOrder[i] != nums[i]) {
                last = i;
                break;
            }
        }
        
        cout << last - first + 1;
        return last - first + 1;
    }
};

int main() {
  vector<int>nums = {2, 6, 4, 8, 10, 9, 15};
  Solution sol;
  sol.findUnsortedSubarray(nums);
}
   

File: shortest_unsorted_continuous_subarray/kth/ShortestUnsortedContinuousSubarray.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp7
{
    class Program
    {
        static void Main(string[] args)
        {
            Debug.WriteLine(new Solution().FindUnsortedSubarray((new int[] { 1,2,3,4 }))); 

        }
    }
    public class Solution
    {
        public int FindUnsortedSubarray(int[] nums)
        {
            var sortedNums = (from num in nums orderby num select num).ToList();
            var diff = sortedNums.Select((num, index) => new { Number = num, Index = index })
                .Where(x => nums[x.Index] != x.Number).Select(x => x.Index).ToList();
            return diff.Count() == 0 ? 0 : diff.Last() - diff.First() + 1;
        }
    }
}
